rbac/server/api/roles.py

```diff
@@ -183,9 +184,20 @@
         request.app.config.TIMEOUT,
         request.json.get("tracker") and True,
     )
-    if request.json.get("tracker"):
-        return utils.create_tracker_response("batch_status", batch_status)
-    return json({"proposal_id": proposal_id})
+    conn = await create_connection()
+    role_resource = await roles_query.fetch_role_resource(conn, role_id)
+    owners = role_resource.get("owners")
+    conn.close()
+    requester_id = request.json.get("id")
+    if requester_id in owners:
+        request.json["status"] = "APPROVED"
+        request.json["reason"] = "I am the owner of this role"
+        await proposals.update_proposal(request, proposal_id)
+        raise ApiBadRequest("Owner is the requester. Proposal is autoapproved")
```

Comment:
This should still return a 200 status code as the ApiBadRequest returns a 400 error code.

Reply:
Yes, there is a UI ticket for that. Ticket # 276